x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -120,7 +120,7 @@ impl Program {
     /// # Panics
     /// Panics if there was an issue deserializing the account
     #[must_use]
-    pub fn deploy(&self, program_id: Id, account_creation_data: &[u8]) -> Address {
+    pub fn deploy(&self, program_id: &[u8], account_creation_data: &[u8]) -> Address {
```

Comment:
Can you just change the `Id` type instead? Then you can make this `&Id` (or the type could still be owned).

Reply:
it would have to be a distinct type since IDs still exist as the result of hashes.